### PR TITLE
ENH: add IntervalProd.__getitem__

### PR DESCRIPTION
**Pull request by adler-j**
Another short pull, would close issue #140.

### CODE REVIEW

File: odl/set/domain.py [hunk 1]
```diff
@@ -608,6 +611,41 @@ def __len__(self):
...
+        Interval(-1.0, -0.5)
+
+        By slice
+
+        >>> rbox[:]
+        Cuboid([-1.0, 2.0, 0.0], [-0.5, 3.0, 0.5])
+        >>> rbox[::2]
+        Rectangle([-1.0, 0.0], [-0.5, 0.5])
+
+        By list of ints
```
Comment (kohr-h) [inline odl/set/domain.py]: Blank line missing

File: odl/set/domain.py [hunk 2]
```diff
@@ -47,6 +47,9 @@ class IntervalProd(Set):
     An interval product is a Cartesian product of n intervals, i.e. an
     n-dimensional rectangular box aligned with the coordinate axes
     as a subset of :math:`R^n`.
+
+    `IntervalProd` objects are immutable, all methods involving the return
```
Comment (kohr-h) [inline odl/set/domain.py]: the what?

**Comment by kohr-h** [comment]
OK

File: odl/set/domain.py (hunk 2 above)
Comment (adler-j) [inline odl/set/domain.py]: them

**Pull request merged**